src/Sulu/Bundle/MediaBundle/Media/FormatLoader/BaseXmlFormatLoader.php

```diff
@@ -0,0 +1,246 @@
+
+/**
+ * Abstract xml loader for the image formats which contains the common part of all versions.
+ */
+abstract class BaseXmlFormatLoader extends FileLoader
+{
+    const XML_NAMESPACE_URI = 'http://schemas.sulu.io/media/formats';
+
+    const SCHEMA_URI = '';
+
+    const SCALE_MODE_DEFAULT = 'outbound';
+
+    /**
+     * @var \DOMXPath
+     */
+    protected $xpath;
+
+    /**
+     * @var array
+     */
+    private $defaultOptions = [];
+
+    /**
+     * @return array
+     */
+    public function getDefaultOptions()
+    {
+        return $this->defaultOptions;
+    }
+
+    /**
+     * @param array $defaultOptions
+     */
+    public function setDefaultOptions($defaultOptions)
+    {
+        $this->defaultOptions = $defaultOptions;
+    }
+
+    /**
+     * Load formats from a xml file.
+     *
+     * @param mixed $resource The resource
+     * @param string $type The resource type
+     *
+     * @return array The formats array for the given resource
+     */
+    public function load($resource, $type = null)
+    {
+        $path = $this->getLocator()->locate($resource);
+
+        return $this->parseXml($path);
+    }
+
+    /**
+     * Returns true if this class supports the given resource.
+     *
+     * @param mixed $resource A resource
+     * @param string $type The resource type
+     *
+     * @return bool true if this class supports the given resource, false otherwise
+     */
+    public function supports($resource, $type = null)
+    {
+        if (!is_string($resource) || 'xml' !== pathinfo($resource, PATHINFO_EXTENSION)) {
+            return false;
+        }
+
+        $file = $this->getLocator()->locate($resource);
+        $document = XmlUtils::loadFile($file);
+        $namespaces = $document->documentElement->attributes->getNamedItem('schemaLocation')->nodeValue;
+
+        $start = strpos($namespaces, static::XML_NAMESPACE_URI) + strlen(static::XML_NAMESPACE_URI) + 1;
+        $namespace = substr($namespaces, $start);
+
+        $end = strpos($namespace, ' ');
+        if ($end !== false) {
+            $namespace = substr($namespace, 0, $end);
+        }
+
+        return $namespace === static::SCHEMA_URI;
+    }
+
+    /**
+     * @param $file
+     *
+     * @return array
+     */
+    private function parseXml($file)
+    {
+        $formats = [];
+
+        // load xml file
+        $xmlDoc = $this->tryLoad($file);
+
+        $this->xpath = new \DOMXPath($xmlDoc);
+        $this->xpath->registerNamespace('x', static::XML_NAMESPACE_URI);
+
+        foreach ($this->xpath->query('/x:formats/x:format') as $formatNode) {
+            $this->addFormatFromFormatNode($formatNode, $formats);
+        }
+
+        return $formats;
+    }
+
+    /**
+     * For a given format node and a given array of formats, this method parses the
+     * format node to an array and adds it to the formats array. If a format with the
+     * same key is already in the passed array, the node does not get parsed or added.
```

Comment:
What about throwing an exception instead? Would make hard to find errors less common.